x-pack/metricbeat/module/gcp/metrics/metrics_requester.go

```diff
@@ -124,6 +124,12 @@ func (r *metricsRequester) Metrics(ctx context.Context, serviceName string, alig
 		}
 	}
 
+	start := time.Now()
+	defer func() {
+		totalTime := time.Since(start)
+		r.logger.Debugf("total time taken for all ListTimeSeries requests: %s config period: %s", totalTime, r.config.period.AsDuration())
+	}()
+
```

Comment:
Same here,

```suggestion
	defer func() {
		s.logger.Debugf("total time taken for compute AggregatedList request: %s", time.Since(time.Now()))
	}()
```